src/go/rpk/pkg/cli/cmd/redpanda/admin/brokers/decommission.go

```diff
@@ -0,0 +1,100 @@
+package brokers
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/api/admin"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/redpanda"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+func newDecommissionBroker(fs afero.Fs) *cobra.Command {
+	var force bool
+	cmd := &cobra.Command{
+		Use:   "decommission [BROKER ID]",
+		Short: "Decommission the given broker",
+		Long: `Decommission the given broker.
+
+Decommissioning a broker removes it from the cluster.
+
+A decommission request is sent to every broker in the cluster, only the cluster
+leader handles the request.
+
+For safety on v22.x clusters, this command will not run if the requested 
+broker is in maintenance mode. As of v23.x, Redpanda supports 
+decommissioning a node that is currently in maintenance mode. If you are on 
+a v22.x cluster and need to bypass the maintenance mode check (perhaps your 
+cluster is unreachable), use the hidden --force flag.
+`,
+		Args: cobra.ExactArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			broker, err := strconv.Atoi(args[0])
+			out.MaybeDie(err, "invalid broker %s: %v", args[0], err)
+			if broker < 0 {
+				out.Die("invalid negative broker id %v", broker)
+			}
+
+			p := config.ParamsFromCommand(cmd)
+			cfg, err := p.Load(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+
+			cl, err := admin.NewClient(fs, cfg)
+			out.MaybeDie(err, "unable to initialize admin client: %v", err)
+
+			if !force {
+				brokers, err := cl.Brokers(cmd.Context())
+				out.MaybeDie(err, "unable to get broker list: %v; to bypass the node version check re-run this with --force", err)
+
+				var (
+					b             admin.Broker
+					found, anyOld bool
+				)
+				for _, br := range brokers {
+					if br.NodeID == broker {
+						version, err := redpanda.VersionFromString(br.Version)
+						out.MaybeDie(err, "unable to get broker %q version: %v; to bypass the node version check re-run this with --force", br.NodeID, err)
+						isOld := version.Less(redpanda.Version{Year: 23, Feature: 1, Patch: 1})
+
+						anyOld = anyOld || isOld
```

Comment:
Just wondering what `anyOld`'s intention is. It's called just once and equivalent to `isOld`, no?
I guess version check should run for every node in `brokers`?

Reply:
Yes, per @bharathv's suggestion, the broker we actually need to check is the controller broker. The controller isn't available in the /brokers response, so instead we'll just simplify how much we need to do and just check every broker.